src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,346 @@
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2 12 $4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2+12$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2(12)$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2( 12)$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2(+12)$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2)$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2$3($4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2)$3($4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2$3$4($5)$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2$3$4$5($6)$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "+ 38$2$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "++38$2$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "+3 8$2$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2 0 50 $4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2 05 0 $4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2$3$4 67 8 $6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2$3$4 6 7 8 $6$7$8$9"));
+        }
+
+        List<String> specialChars = getSpecialCharacters("\\");
+        specialChars.removeAll(List.of("\\.", "\\-"));
+
+        specialChars.forEach(
+                specialChar -> wrongPhoneNumbers.forEach(
+                        number -> {
+                            tempList.add(
+                                    number.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN,
+                                            "$1$2$3$4$5".concat(specialChar).concat("$7$8$9")));
+                            String twoDigits = number.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$7");
+                            tempList.add(
+                                    number.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN,
+                                            "$1$2$3$4$5$6".concat("\\".concat(twoDigits.replaceAll("^\\d", specialChar))).concat("$8$9")));
+                        }));
+
+        wrongPhoneNumbers.clear();
+        wrongPhoneNumbers.addAll(tempList);
+
+        return wrongPhoneNumbers;
+    }
+
+    private static List<String> createCorrectUserNames() {
+        List<String> correctNames = new ArrayList<>(List.of(
+                "pushkin",
+                "ThisNameHasLengthMoreThan100SymbolsThisNameHasLengthMoreThan100SymbolsThisNameHasLengthMoreThan100Symbols",
```

Comment:
Why don't you just used IntStream.range with size more than 100 symbols and fill it with any char?